dashboard/app/controllers/sections_controller.rb

```diff
@@ -24,13 +23,24 @@ def update
       course_id ||= script.course.try(:id)
     end
 
-    section.update!(course_id: course_id, script_id: script_id)
+    # TODO: (madelynkasula) refactor to use strong params
+    fields = {}
+    fields[:course_id] = course_id
+    fields[:script_id] = script_id
```

Comment:
it looks like we are no longer checking that the course and script ids are valid. is this something we are expecting rails validation to do for us? my understanding is that it does, but you two probably know better than I do.

Reply:
no, i think you're right. i had left `course_id` and `script_id` alone since they were already apart of this endpoint, but we probably should be doing those checks as they were in pegasus:

https://github.com/code-dot-org/code-dot-org/blob/419502aed4dc98b481590f5ef614833521541bb9/pegasus/helpers/section_api_helpers.rb#L687-L705

i'll update this PR with those checks 👍

Reply:
This blog post suggests this validation is done for us as of Rails 5: https://blog.bigbinary.com/2016/02/15/rails-5-makes-belong-to-association-required-by-default.html

One approach is the write the test first, asserting that the Section#update fails without a valid course or script id. if it passes without any code changes, then you're done :-)

Reply:
🤦‍♀️ there are already tests for this i hadn't seen before:

https://github.com/code-dot-org/code-dot-org/blob/88ac044611e411d8d6c84ee3fa9a7ccdbaa337b0/dashboard/test/controllers/api/v1/sections_controller_test.rb#L416-L427

and

https://github.com/code-dot-org/code-dot-org/blob/88ac044611e411d8d6c84ee3fa9a7ccdbaa337b0/dashboard/test/controllers/api/v1/sections_controller_test.rb#L442-L455

so we should be covered!